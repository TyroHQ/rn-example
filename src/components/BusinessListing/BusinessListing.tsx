import React from "react";
import { View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";
import { BusinessListingHeader } from "../BusinessListingHeader";
import { PointStatusBox } from "../PointStatusBox";
import { RewardTileHeader } from "../RewardTileHeader";

const BusinessListingContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class BusinessListing extends React.Component<{}> {
  static defaultProps: BusinessListing["props"];
  render() {
    return (
      <Placeholder>
        Business Expanded Container
        <BusinessListingHeader />
        <Placeholder>
          Dropdown white Container
          <Placeholder>
            Point status Boxes
            <PointStatusBox />
            <PointStatusBox />
            <PointStatusBox />
          </Placeholder>
          <RewardTileHeader />
          <Placeholder>
            White Buttons
            <Placeholder>Business info button</Placeholder>
            <Placeholder>Reward Rules button</Placeholder>
            <Placeholder>Notifications button</Placeholder>
          </Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
