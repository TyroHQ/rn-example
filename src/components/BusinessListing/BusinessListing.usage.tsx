import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { BusinessListing } from "./";

export const basic = () => <BusinessListing />;

export const advanced = () => <BusinessListing />;
