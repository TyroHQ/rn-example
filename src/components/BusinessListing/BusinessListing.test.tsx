import { shallow } from "enzyme";
import * as usage from "./BusinessListing.usage";

describe("<BusinessListing />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
