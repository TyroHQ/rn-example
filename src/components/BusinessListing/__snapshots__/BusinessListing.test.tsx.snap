// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessListing /> renders well in advanced mode 1`] = `
<Placeholder
  orientation="row"
>
  Business Expanded Container
  <BusinessListingHeader />
  <Placeholder
    orientation="row"
  >
    Dropdown white Container
    <Placeholder
      orientation="row"
    >
      Point status Boxes
      <PointStatusBox />
      <PointStatusBox />
      <PointStatusBox />
    </Placeholder>
    <RewardTileHeader />
    <Placeholder
      orientation="row"
    >
      White Buttons
      <Placeholder
        orientation="row"
      >
        Business info button
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Reward Rules button
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Notifications button
      </Placeholder>
    </Placeholder>
  </Placeholder>
</Placeholder>
`;

exports[`<BusinessListing /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Business Expanded Container
  <BusinessListingHeader />
  <Placeholder
    orientation="row"
  >
    Dropdown white Container
    <Placeholder
      orientation="row"
    >
      Point status Boxes
      <PointStatusBox />
      <PointStatusBox />
      <PointStatusBox />
    </Placeholder>
    <RewardTileHeader />
    <Placeholder
      orientation="row"
    >
      White Buttons
      <Placeholder
        orientation="row"
      >
        Business info button
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Reward Rules button
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Notifications button
      </Placeholder>
    </Placeholder>
  </Placeholder>
</Placeholder>
`;
